lib/modules/manager/bun/artifacts.ts

```diff
@@ -0,0 +1,86 @@
+import is from '@sindresorhus/is';
+import { TEMPORARY_ERROR } from '../../../constants/error-messages';
+import { logger } from '../../../logger';
+import { exec } from '../../../util/exec';
+import type { ExecOptions } from '../../../util/exec/types';
+import {
+  deleteLocalFile,
+  readLocalFile,
+  writeLocalFile,
+} from '../../../util/fs';
+import type { UpdateArtifact, UpdateArtifactsResult } from '../types';
+
+export async function updateArtifacts(
+  updateArtifact: UpdateArtifact
+): Promise<UpdateArtifactsResult[] | null> {
+  const { packageFileName, updatedDeps, newPackageFileContent, config } =
+    updateArtifact;
+  logger.debug({ updateArtifact }, `bun.updateArtifacts(${packageFileName})`);
+  const isLockFileMaintenance = config.updateType === 'lockFileMaintenance';
+
+  if (is.emptyArray(updatedDeps) && !isLockFileMaintenance) {
+    logger.debug('No updated bun deps - returning null');
+    return null;
+  }
+
+  const lockFileName = config.lockFiles?.[0];
+
+  if (!lockFileName) {
+    logger.debug(`No ${lockFileName} found`);
+    return null;
+  }
+
+  const oldLockFileContent = await readLocalFile(lockFileName);
+  if (!oldLockFileContent) {
+    logger.debug(`No ${lockFileName} found`);
+    return null;
+  }
+
+  try {
+    await writeLocalFile(packageFileName, newPackageFileContent);
```

Comment:
Isn't this already done on a upper layer? i don't see this on other managers. 🤔

Reply:
![image](https://github.com/renovatebot/renovate/assets/6311784/e6095a04-90de-4809-90d4-5ffdf547874a)